core/dbt/deps/tarball.py

```diff
@@ -42,23 +48,31 @@ def nice_version_name(self):
         return f"tarball (url: {self.tarball})"
 
     def _fetch_metadata(self, project, renderer):
-        """
-        recycle RegistryPackageMetadata so that we can use the install and
-        download_and_untar from RegistryPinnedPackage next.
-        build RegistryPackageMetadata from info passed via packages.yml since no
-        'metadata' service exists in this case.
-        """
-
-        dct = {
-            "name": self.package,
```

Comment:
In theory, if we're downloading the package to read its metadata, it would no longer be required for users to provide the package `name`

Reply:
Legit! Okay, let's call this out of scope for the present PR. @emmyoop made a really good point that requiring the user to type the `name` in `packages.yml` is probably better for readability anyway.